internal/controller/operatorconfigmap_controller.go

```diff
@@ -469,52 +469,89 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI(storageClients *v1al
 				DomainLabels: slices.Collect(maps.Keys(topologyDomainLablesSet)),
 			}
 		}
+		driverSpecDefaults.GenerateOMapInfo = ptr.To(c.shouldGenerateRBDOmapInfo())
 		return nil
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile csi operator config: %v", err)
 	}
 
+	// get the custom csi Tolerations
+	// TODO: Remove it after 4.19 release
+	customNodePluginTolerations, customControllerPluginTolerations, err := c.getCsiCustomTolerations()
+	if err != nil {
+		return fmt.Errorf("failed to get custom tolerations for csi: %v", err)
+	}
+
 	// ceph rbd driver config
 	rbdDriver := &csiopv1a1.Driver{}
 	rbdDriver.Name = templates.RBDDriverName
 	rbdDriver.Namespace = c.OperatorNamespace
+	rbdDriver.Spec = csiopv1a1.DriverSpec{
+		NodePlugin: &csiopv1a1.NodePluginSpec{
+			PodCommonSpec: csiopv1a1.PodCommonSpec{
+				Tolerations: customNodePluginTolerations,
+			},
+		},
+		ControllerPlugin: &csiopv1a1.ControllerPluginSpec{
+			PodCommonSpec: csiopv1a1.PodCommonSpec{
+				Tolerations: customControllerPluginTolerations,
+			},
+		},
+	}
 
-	if err := c.createOrUpdate(rbdDriver, func() error {
-		if err := c.own(rbdDriver); err != nil {
-			return fmt.Errorf("failed to own csi rbd driver: %v", err)
-		}
-		rbdDriver.Spec.GenerateOMapInfo = ptr.To(c.shouldGenerateRBDOmapInfo())
-		return nil
-	}); err != nil {
-		return fmt.Errorf("failed to reconcile rbd driver: %v", err)
+	if err := c.Create(c.ctx, rbdDriver); err != nil && !kerrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create csi rbd driver: %v", err)
+	}
+	if err := c.own(rbdDriver); err != nil {
+		return fmt.Errorf("failed to own csi rbd driver: %v", err)
 	}
 
 	// ceph fs driver config
 	cephFsDriver := &csiopv1a1.Driver{}
 	cephFsDriver.Name = templates.CephFsDriverName
 	cephFsDriver.Namespace = c.OperatorNamespace
+	cephFsDriver.Spec = csiopv1a1.DriverSpec{
+		NodePlugin: &csiopv1a1.NodePluginSpec{
+			PodCommonSpec: csiopv1a1.PodCommonSpec{
+				Tolerations: customNodePluginTolerations,
+			},
+		},
+		ControllerPlugin: &csiopv1a1.ControllerPluginSpec{
+			PodCommonSpec: csiopv1a1.PodCommonSpec{
+				Tolerations: customControllerPluginTolerations,
+			},
+		},
+	}
 
-	if err := c.createOrUpdate(cephFsDriver, func() error {
-		if err := c.own(cephFsDriver); err != nil {
-			return fmt.Errorf("failed to own csi cephfs driver: %v", err)
-		}
-		return nil
-	}); err != nil {
-		return fmt.Errorf("failed to reconcile cephfs driver: %v", err)
+	if err := c.Create(c.ctx, cephFsDriver); err != nil && !kerrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create csi cephfs driver: %v", err)
+	}
```

Comment:
Why not `createOrUpdate` (because you always need to do the own part)? then use the UID to know if you are creating or updating?

Reply:
Prone to errors, as we might by mistake can add spec to update.